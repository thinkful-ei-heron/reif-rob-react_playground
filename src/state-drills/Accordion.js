import React, { Component } from 'react'

export default class Accordion extends Component {
  static defaultProps = {
    sections: []
  }

  state = {
    lastClicked: null
  }

  clickHandler = (index) => {
    this.setState({
      lastClicked: index
    })
  }

  createListItem = (item, index) => {
    return (
      <li className='accordion-item' key={index}>
        <button onClick={() => this.clickHandler(index)} ref='itemButton'>{item.title}</button>
        {(this.state.lastClicked === index) && <p>{item.content}</p>}
      </li>
    )
  }

  render() {
    return (
      <ul className="sections-container">
        {/* li element for each item in props.sections */}
        {/* <button>item.title</button> */}
        {/* onClick -> render <p>item.content</p> below */}
        {/* only one <p> visible at a time */}
        {this.props.sections.map((item, index) =>
          this.createListItem(item, index)
        )}
      </ul>
    )
  }
}
