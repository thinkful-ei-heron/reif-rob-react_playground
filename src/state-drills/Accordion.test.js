import React from 'react';
import renderer from 'react-test-renderer';

import { shallow } from 'enzyme'
import toJson from 'enzyme-to-json'

import Accordion from './Accordion';

const sections = [
  {
    title: 'Section 1',
    content: 'Lorem ipsum dolor sit amet consectetur adipisicing elit.',
  },
  {
    title: 'Section 2',
    content: 'Cupiditate tenetur aliquam necessitatibus id distinctio quas nihil ipsam nisi modi!',
  },
  {
    title: 'Section 3',
    content: 'Animi amet cumque sint cupiditate officia ab voluptatibus libero optio et?',
  },
]

describe('<ul>', () => {
  it('renders empty list', () => {
    const body = renderer.create(<Accordion />).toJSON();
    expect(body).toMatchSnapshot()
  });

  it('renders no sections as active by default', () => {
    const body = renderer.create(<Accordion sections={sections} />).toJSON();
    expect(body).toMatchSnapshot()
  });

  it('Opens contents when clicked', () => {
    const body = shallow(<Accordion sections={sections} />);
    body.find('button').at(1).simulate('click');
    expect(toJson(body)).toMatchSnapshot()
  });

  it('Only one content shown', () => {
    const body = shallow(<Accordion sections={sections} />);
    body.find('button').at(0).simulate('click');
    body.find('button').at(1).simulate('click');
    expect(toJson(body)).toMatchSnapshot()
  });
})