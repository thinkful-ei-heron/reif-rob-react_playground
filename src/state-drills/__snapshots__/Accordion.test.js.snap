// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ul> Only one content shown 1`] = `
<ul
  className="sections-container"
>
  <li
    className="accordion-item"
    key="0"
  >
    <button
      onClick={[Function]}
    >
      Section 1
    </button>
  </li>
  <li
    className="accordion-item"
    key="1"
  >
    <button
      onClick={[Function]}
    >
      Section 2
    </button>
    <p>
      Cupiditate tenetur aliquam necessitatibus id distinctio quas nihil ipsam nisi modi!
    </p>
  </li>
  <li
    className="accordion-item"
    key="2"
  >
    <button
      onClick={[Function]}
    >
      Section 3
    </button>
  </li>
</ul>
`;

exports[`<ul> Opens contents when clicked 1`] = `
<ul
  className="sections-container"
>
  <li
    className="accordion-item"
    key="0"
  >
    <button
      onClick={[Function]}
    >
      Section 1
    </button>
  </li>
  <li
    className="accordion-item"
    key="1"
  >
    <button
      onClick={[Function]}
    >
      Section 2
    </button>
    <p>
      Cupiditate tenetur aliquam necessitatibus id distinctio quas nihil ipsam nisi modi!
    </p>
  </li>
  <li
    className="accordion-item"
    key="2"
  >
    <button
      onClick={[Function]}
    >
      Section 3
    </button>
  </li>
</ul>
`;

exports[`<ul> renders empty list 1`] = `
<ul
  className="sections-container"
/>
`;

exports[`<ul> renders no sections as active by default 1`] = `
<ul
  className="sections-container"
>
  <li
    className="accordion-item"
  >
    <button
      onClick={[Function]}
    >
      Section 1
    </button>
  </li>
  <li
    className="accordion-item"
  >
    <button
      onClick={[Function]}
    >
      Section 2
    </button>
  </li>
  <li
    className="accordion-item"
  >
    <button
      onClick={[Function]}
    >
      Section 3
    </button>
  </li>
</ul>
`;
